onnxruntime/core/providers/common.h

```diff
@@ -42,4 +42,42 @@ inline float clamp(float v, float lo, float hi) {
   return v;
 }
 
+/**
+Tries to call the given function parallelly, splitted into specified batch(es)
+**/
+template <typename F>
+inline void TryBatchParallelFor(concurrency::ThreadPool* tp, int32_t total, F&& fn, int32_t batch_size = 0) {
```

Comment:
LGTM. But,  I think it could be better if you can reuse this function: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/platform/threadpool.cc#L166

Reply:
in which case will it overflow?

Reply:
this function does what exactly I need, but seems not accessible from `concurrency::ThreadPool`.